zcash_client_sqlite/src/wallet/init/migrations/tx_retrieval_queue.rs

```diff
@@ -133,10 +135,89 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
 
 #[cfg(test)]
 mod tests {
-    use crate::wallet::init::migrations::tests::test_migrate;
+    use rusqlite::named_params;
+    use secrecy::Secret;
+    use tempfile::NamedTempFile;
+    use zcash_primitives::{
+        legacy::{Script, TransparentAddress},
+        transaction::{components::transparent, Authorized, TransactionData, TxVersion},
+    };
+    use zcash_protocol::{
+        consensus::{BranchId, Network},
+        value::Zatoshis,
+    };
+
+    use crate::{
+        wallet::init::{init_wallet_db_internal, migrations::tests::test_migrate},
+        WalletDb,
+    };
+
+    use super::{DEPENDENCIES, MIGRATION_ID};
 
     #[test]
     fn migrate() {
-        test_migrate(&[super::MIGRATION_ID]);
+        test_migrate(&[MIGRATION_ID]);
+    }
+
+    #[test]
+    fn migrate_with_data() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), Network::TestNetwork).unwrap();
+
+        let seed_bytes = vec![0xab; 32];
+
+        // Migrate to database state just prior to this migration.
+        init_wallet_db_internal(
+            &mut db_data,
+            Some(Secret::new(seed_bytes.clone())),
+            &DEPENDENCIES,
```

Comment:
This is a good way to approach migration tests; it ensures that _only_ the nodes in the graph leading up to this node are evaluated, so anything that's missing should get caught. It does require inspecting what data is required in order to test all the possible branches within the migration though.